src/widgets/NavigationBar.js

```diff
@@ -8,38 +8,78 @@ import PropTypes from 'prop-types';
 import { Platform, TouchableOpacity, View, StyleSheet } from 'react-native';
 import Icon from 'react-native-vector-icons/FontAwesome';
 import { textStyles } from '../globalStyles';
+import { BadgeSet } from './BadgeSet';
 
-export function NavigationBar({ onPressBack, LeftComponent, CentreComponent, RightComponent }) {
-  return (
-    <View style={localStyles.container} >
-      <View style={localStyles.leftSection}>
-        <TouchableOpacity onPress={onPressBack} style={localStyles.backButton}>
-          {onPressBack && <Icon name={'chevron-left'} style={localStyles.backIcon} />}
-        </TouchableOpacity>
-        {LeftComponent && <LeftComponent />}
-      </View>
-      <View style={localStyles.centreSection} >
-        {CentreComponent && <CentreComponent />}
-      </View>
-      <View style={localStyles.rightSection}>
-        {RightComponent && <RightComponent />}
+export class NavigationBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+    };
+  }
+
+  componentWillReceiveProps(props) {
+    const dataType = this.getDataTypeFromTitle(props);
+    if (dataType) this.refreshData(dataType);
+  }
+
+  getDataTypeFromTitle(props) {
+    switch (props.currentTitle) {
+      case 'Customer Requisitions':
```

Comment:
Is this translation safe? You could use `navStrings.<the page title>` but that's still not ideal. Is routeName available in this scope? https://github.com/openmsupply/mobile/blob/%23507-Badge-highlight-number/src/pages/PageContainer.js#L36

Reply:
Resolved with b75124e